File: src/Entity/Question.php
<?php

namespace App\Entity;

use App\Repository\QuestionRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use App\Entity\Formulaire;

#[ORM\Entity(repositoryClass: QuestionRepository::class)]
class Question
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le texte de la question est obligatoire.")]
    private ?string $questionText = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le type de question est obligatoire.")]
    private ?string $type = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La réponse correcte est obligatoire.")]
    private ?string $correctAnswer = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le nombre de points est obligatoire.")]
    #[Assert\Positive(message: "Les points doivent être un nombre positif.")]
    private ?int $points = null;

    #[ORM\ManyToOne(targetEntity: Formulaire::class, inversedBy: "questions")]
    #[ORM\JoinColumn(nullable: false)]
    private ?Formulaire $formulaire = null;

    public function getId(): ?int { return $this->id; }
    public function getQuestionText(): ?string { return $this->questionText; }
    public function setQuestionText(string $questionText): static { $this->questionText = $questionText; return $this; }
    public function getType(): ?string { return $this->type; }
    public function setType(string $type): static { $this->type = $type; return $this; }
    public function getCorrectAnswer(): ?string { return $this->correctAnswer; }
    public function setCorrectAnswer(string $correctAnswer): static { $this->correctAnswer = $correctAnswer; return $this; }
    public function getPoints(): ?int { return $this->points; }
    public function setPoints(int $points): static { $this->points = $points; return $this; }
    public function getFormulaire(): ?Formulaire { return $this->formulaire; }
    public function setFormulaire(?Formulaire $formulaire): static { $this->formulaire = $formulaire; return $this; }
}

File: src/main/java/models/Question.java
package models;

public class Question {
    private int id;
    private String questionText;
    private String correctAnswer;
    private int points;
    private String type;
    private int formulaireId;

    public Question() {}

    public Question(String questionText, String correctAnswer, int points, String type, int formulaireId) {
        this.questionText = questionText;
        this.correctAnswer = correctAnswer;
        this.points = points;
        this.type = type;
        this.formulaireId = formulaireId;
    }

    public Question(int id, String questionText, String correctAnswer, int points, String type, int formulaireId) {
        this.id = id;
        this.questionText = questionText;
        this.correctAnswer = correctAnswer;
        this.points = points;
        this.type = type;
        this.formulaireId = formulaireId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getQuestionText() {
        return questionText;
    }

    public void setQuestionText(String questionText) {
        this.questionText = questionText;
    }

    public String getCorrectAnswer() {
        return correctAnswer;
    }

    public void setCorrectAnswer(String correctAnswer) {
        this.correctAnswer = correctAnswer;
    }

    public int getPoints() {
        return points;
    }

    public void setPoints(int points) {
        this.points = points;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getFormulaireId() {
        return formulaireId;
    }

    public void setFormulaireId(int formulaireId) {
        this.formulaireId = formulaireId;
    }

    @Override
    public String toString() {
        return "Question{" +
                "id=" + id +
                ", questionText='" + questionText + '\'' +
                ", points=" + points +
                ", type='" + type + '\'' +
                ", formulaireId=" + formulaireId +
                '}';
    }
}
